app/ante/evm/fee_checker.go

```diff
@@ -35,15 +35,10 @@ import (
 // a) feeCap = tx.fees / tx.gas
 // b) tipFeeCap = tx.MaxPriorityPrice (default) or MaxInt64
 // - when `ExtensionOptionDynamicFeeTx` is omitted, `tipFeeCap` defaults to `MaxInt64`.
-// - when london hardfork is not enabled, it fallbacks to SDK default behavior (validator min-gas-prices).
+// - when london hardfork is not enabled, it falls back to SDK default behavior (validator min-gas-prices).
 // - Tx priority is set to `effectiveGasPrice / DefaultPriorityReduction`.
-func NewDynamicFeeChecker(k DynamicFeeEVMKeeper) authante.TxFeeChecker {
-	return func(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
-		feeTx, ok := tx.(sdk.FeeTx)
-		if !ok {
-			return nil, 0, fmt.Errorf("tx must be a FeeTx")
-		}
```

Comment:
@fedekunze Q: is OK removing this check? because if at runtime we get a Tx that is not implementing the `sdk.FeeTx`, it will panic

Reply:
Afaik all the txs implement the fee tx interface, but we should check somewhere in the ante handler that they do

Reply:
It's checked in the AnteHandler decorator

```
// AnteHandle ensures that the transaction contains valid fee requirements and tries to deduct those
// from the account balance or unclaimed staking rewards, which the transaction sender might have.
func (dfd DeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
	feeTx, ok := tx.(sdk.FeeTx)
	if !ok {
		return ctx, errorsmod.Wrap(errortypes.ErrTxDecode, "Tx must be a FeeTx")
	}
```